aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/LastSyncMetadata.java

```diff
@@ -67,22 +83,23 @@ private LastSyncMetadata(String id, String modelClassName, Long lastSyncTime) {
      */
     static <T extends Model> LastSyncMetadata neverSynced(@NonNull Class<T> modelClass) {
         Objects.requireNonNull(modelClass);
-        return create(modelClass, null);
+        return create(modelClass, null, SyncType.BASE);
```

Comment:
Are these two `LastSyncMetadata` `equals()`? Should they have the same `hashCode()`?

1. Person.class, null, SyncType.BASE
2. Person.class, null, SyncType.DELTA

Might be an edge case / not super relevant.

Reply:
Interesting...Right now they're treated as not equals, which I think is OK. I ran the following test to verify that:

```
        LastSyncMetadata data1 = LastSyncMetadata.create(BlogOwner.class, null, SyncType.BASE);
        LastSyncMetadata data2 = LastSyncMetadata.create(BlogOwner.class, null, SyncType.DELTA);
        assertNotEquals(data1, data2);
        assertNotEquals(data1.hashCode(), data2.hashCode());
```